### PR TITLE
Update token renewal poll timing & remove JMAP basic auth support 

### PR DESCRIPTION
**Pull request by samiulsami**

### CODE REVIEW

File: inbox/clients.go [hunk 1]
```diff
@@ -196,56 +189,53 @@ func (j *JMAPClient) RenewSession() error {
...
+				errs = append(errs, fmt.Errorf("failed to initialize mailbox ids: %w", err))
 				return false, nil
 			}
 
 			return true, nil
 		})
 	if err != nil {
-		return errors.Wrap(lastErr, err.Error())
+		errs = append(errs, err)
+		return fmt.Errorf("failed to renew JMAP Session: %s", go_errs.Join(errs...))
```
Comment (Imtiaz246) [inline inbox/clients.go]: If its really a failure the error msg will very big. We don't actually need to save the error. We can simply take the error returned by the wait.Poll... and wrap it with "couldn't get inbox service token" or "error getting inbox service token". Just use the returned error

File: inbox/clients.go [hunk 2]
```diff
@@ -105,16 +106,16 @@ func (w *WebAdminClient) refreshRequired() bool {
...
 
 func (w *WebAdminClient) renew() error {
-	var lastErr error
+	errs := make([]error, 0)
 	err := wait.PollUntilContextTimeout(
 		context.TODO(),
-		100*time.Millisecond,
-		5*time.Second,
+		3*time.Second,
+		1*time.Minute,
```
Comment (Imtiaz246) [inline inbox/clients.go]: err := wait.PollUntilContextTimeout(context.TODO(), 3*time.Second, 1*time.Minute, true,
		func(ctx context.Context) (bool, error) {
I think it's better to write this function in two lines.

File: inbox/clients.go [hunk 3]
```diff
@@ -131,20 +132,18 @@ func (w *WebAdminClient) renew() error {
...
 type JMAPClient struct {
 	jmap.Client
-	muJmap     sync.RWMutex
-	userId     jmap.ID
-	userEmail  string
-	mailboxIds map[string]jmap.ID
-
-	forceBasicAuth              bool
-	basicAuthCreds              BasicAuthCredentials
+	muJmap                      sync.RWMutex
```
Comment (Imtiaz246) [inline inbox/clients.go]: name it just mu.

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

File: inbox/clients.go (hunk 1 above)
Comment (samiulsami) [reply inbox/clients.go]: I guess then it makes sense to revert it back to what it was before.

**Review by Imtiaz246 — DISMISSED** [review]

**Pull request merged**